tests/integration/conftest.py

```diff
@@ -439,20 +442,19 @@ def workflow_source(mod_flow, mod_one_conf, tmp_path):
     Yields:
         Path to source directory.
     """
-    def _inner(cfg=None):
+    def _inner(cfg):
         src_dir = _make_src_flow(tmp_path, cfg)
         return src_dir
     yield _inner
 
 
 @pytest.fixture
-def cylc_install(test_dir, run_dir):
+def install(test_dir, run_dir):
     """Install a workflow from source"""
     def _inner(source, **kwargs):
-        Opts = Options(install_gop())
-        opts = Opts(**kwargs)
+        opts = InstallOpts(**kwargs)
         opts.workflow_name = str(test_dir.relative_to(run_dir))
-        install(opts, str(source))
+        workflow_id = cylc_install(opts, str(source))
 
-        return test_dir
+        return workflow_id
```

Comment:
I think this will break if you try to install two workflows in the same test (as they will try to install into the same run dir).

I previously suggested adding a UUID to the workflow ID. On reflection it would probably make more sense to just add `basename source` onto the end something like this:

```python
opts.workflow_name = str(test_dir.relative_to(run_dir) / run_dir.name)
```

E.G. consider the test:

```python
def test_scan(src_dir, one_conf, install):
    foo_src = src_dir(one_conf)
    bar_src = src_dir(one_conf)
    foo_id = install(foo)
    bar_id = install(bar)
    assert listify(scan()) == {foo_id, bar_id}
```